base-theme/layouts/partials/local_video_player.html

```diff
@@ -0,0 +1,34 @@
+{{ $random := delimit (shuffle (split (md5 "seed") "" )) "" }}
+{{ $playerId := printf "video-player-%s-%s" .youtubeKey $random }}
+<div class="video-container embedded-video-container">
+  <video
+    id="{{ $playerId }}"
+    class="video-js vjs-default-skin vjs-big-play-centered vjs-16-9"
+    controls
+    data-setup='{
+    "fluid": true,
+    "inactivityTimeout": 0}'
+  >
+      {{ if .archiveUrl }}
+        <source src="{{ partial "resource_url.html" (dict "context" .context "url" .archiveUrl) }}" type="video/mp4" />
+      {{ end }}
+  </video>
+</div>
+
+{{ if and .startTime .endTime }}
+<script>
+  window.onload = function() {
```

Comment:
The current implementation here doesn't work with the `or` suggestion above. Also, the player is a bit awkward if you press play AFTER `endTime`. (it plays for a second, then pauses). Maybe that's reasonable behavior, though.


1. use `const`
2. conditional on start time / end time
3. **MAYBE, MAYBE NOT**  remove "stopper" after its first application. *The youtube player handles this very nice: The video stops, the play button goes away. You can force the youtube player to continue by scrubbing if you really want.*

Re (3) ... The current behavior, 

```js
  window.onload = function() {
    if (window.videojs) {
      const player = window.videojs("{{ $playerId }}")
      const startTime = {{ .startTime }}
      const endTime = {{ .endTime }}
      if (startTime) {
        player.currentTime(startTime)
      }
      const stopper = () => {
        if (player.currentTime() >= endTime) {
          player.pause()
          player.off("timeupdate", stopper) // Maybe???
        }
      }
      if (endTime) {
        player.on("timeupdate", stopper)
      }
    }
  }
```